Yosemite/Yosemite/Stores/ProductTagStore.swift

```diff
@@ -80,11 +76,12 @@ private extension ProductTagStore {
 
             switch result {
             case .success(let productTags):
-                if pageNumber == Default.firstPageNumber {
-                    self?.deleteUnusedStoredProductTags(siteID: siteID)
-                }
-
-                self?.upsertStoredProductTagsInBackground(productTags, siteID: siteID) {
+                let shouldRemoveUnusedTags = pageNumber == Default.firstPageNumber
+                /// Only upsert tags that are associated with at least 1 product.
+                let filteredTags = productTags.filter { $0.count > 0 }
```

Comment:
> I fixed this by parsing the count property for product tags and use it to filter only used tags when syncing. This ensures that only used tags are recommended, similar to how the feature works in WooCommerce Core.

@itsmeichigo I didn't manage to review the rest of the PR yet, but I'm just leaving this remark to get it discussed, in Core, I noticed that tags with `count = 0` are still suggested, as you can see from the following screenshots:

<img width="937" alt="Screenshot 2024-11-27 at 17 08 41" src="https://github.com/user-attachments/assets/29106754-7491-45a5-8aa7-e23e363ba101">
<img width="280" alt="Screenshot 2024-11-27 at 17 08 55" src="https://github.com/user-attachments/assets/f9090293-d344-4258-9a44-66dc88d7aa89">

Can you double check on your side? I'm not sure what's different between our testing environments

Reply:
What I meant by the tags not being suggested is that it's not listed below the text field (similar to when you select Choose from the most used tags). If you think it's not necessary and that we should display them regardless, I can revert the change. The code to remove unused tags in the storage would be redundant and need to be removed as well.

Reply:
> What I meant by the tags not being suggested is that it's not listed below the text field

I'm not sure I understand what you mean here? the tags listed under the text field in Core are the ones that are already applied to this product, or do you mean something else.

> similar to when you select Choose from the most used tags

Personally I think the app's UX is more aligned with the text field of core than this button, we don't offer a similar thing.

> If you think it's not necessary and that we should display them regardless, I can revert the change.

I would vote to do this yes, but it's not a blocker for the PR, so I'll leave the final decision to you. And just for information, the Android app suggests those tags too, so if we revert this change, we'll keep all platforms aligned.

> The code to remove unused tags in the storage would be redundant and need to be removed as well.

True.

------ 
One related point, if you end up reverting the code, maybe we can sort the tags by their count in the app, this way I think we'll have a UX that offers an experience similar to what Core offers with its widget (I mean the fact that it has an autocomplete text field in addition to the "Choose from the most used tags" button), WDYT? This doesn't need to be done on this PR though.

Reply:
Sorting tags by count sound like a good idea - we'd need to persist the count of products for each tag in the storage to do that. For now, I've reverted the irrelevant changes to make it easier to review this PR - please take another look and let me know what you think - thanks!